### PR TITLE
[CI] Small fixes for DeepSpeed.

### PR DESCRIPTION
**Pull request by tchaton**
## What does this PR do?

Fixes #<issue_number>

## Before submitting
- [ ] Was this discussed/approved via a GitHub issue? (not for typos and docs)
- [ ] Did you read the [contributor guideline](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/.github/CONTRIBUTING.md), **Pull Request** section?
- [ ] Did you make sure your PR does only one thing, instead of bundling different changes together?
- [ ] Did you make sure to update the documentation with your changes? (if necessary)
- [ ] Did you write any new necessary tests? (not for typos and docs)
- [ ] Did you verify new and existing tests pass locally with your changes?
- [ ] Did you update the [CHANGELOG](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/CHANGELOG.md)? (not for typos, docs, test updates, or internal minor changes/refactorings)

## PR review
Anyone in the community is free to review the PR once the tests have passed.
Before you start reviewing make sure you have read [Review guidelines](https://github.com/PyTorchLightning/pytorch-lightning/wiki/Review-guidelines). In short, see the following bullet-list:

 - [ ] Is this pull request ready for review? (if not, please submit in draft mode)
 - [ ] Check that all items from **Before submitting** are resolved
 - [ ] Make sure the title is self-explanatory and the description concisely explains the PR
 - [ ] Add labels and milestones (and optionally projects) to the PR so it can be classified

## Did you have fun?
Make sure you had fun coding 🙃

### CODE REVIEW

File: pytorch_lightning/plugins/training_type/deepspeed.py [hunk 1]
```diff
@@ -316,7 +316,7 @@ def model_sharded_context(self) -> Generator[None, None, None]:
         else:
             model_parallel_context = super().model_sharded_context()
 
-        with model_parallel_context:
+        with torch.cuda.amp.autocast(), model_parallel_context:
```
Comment (SeanNaren) [inline pytorch_lightning/plugins/training_type/deepspeed.py:319]: I think for ZeRO 2/3, autocast isn't used as we rely on a custom DeepSpeed FP16Optimizer to handle this, which converts the entire model to half iirc.

Comment (tchaton) [reply pytorch_lightning/plugins/training_type/deepspeed.py:319]: With newer DeepSpeed version, they check the weights are created in half precision and currently fails in CI.

Comment (SeanNaren) [reply pytorch_lightning/plugins/training_type/deepspeed.py:319]: oh very interesting! i'll check myself when I get time, do they throw an error if the weights are in half precision or not in half precision?

**Comment by SeanNaren** [comment]
If you go through the special tests (for some reason they pass even though a test fails) there still is a failing test. I've removed a line in #7841 and will debug there, as I'm unsure why the changes in this PR are necessary and need more information since the version of DeepSpeed shouldn't have changed

**Comment by carmocca** [comment]
> If you go through the special tests (for some reason they pass even though a test fails) there still is a failing test.

@awaelchli you looked at this already somewhere else, right?

**Comment by awaelchli** [comment]
Yes here #7790

**Pull request closed without merging**